Source/CarthageKit/Dependency.swift

```diff
@@ -4,14 +4,32 @@ import Tentacle
 
 /// Uniquely identifies a project that can be used as a dependency.
 public enum Dependency {
+	/// Uniquely identifies a Binary Spec's resolved URL and its description
+	public struct BinaryURL: CustomStringConvertible {
+		/// A Resolved URL
+		public let url: URL
+
+		/// An Optional custom description
+		public let overloadedDescription: String?
+
+		public var description: String {
+			return overloadedDescription ?? url.description
+		}
+
+		init(url: URL, overloadedDescription: String? = nil) {
+			self.url = url
+			self.overloadedDescription = overloadedDescription
+		}
```

Comment:
Do we need to get the absolute URL when reading from `Cartfile.resolved`? Will it cause problems if this binary URL isn't equal to the one that's read from the `Cartfile`?

Reply:
That is one thing I wasn't sure of.

My expectation was:
If Cartfile uses a relative path, so should a Cartfile.resolved, otherwise, an absolute file page would break between machines. Otherwise, it would appear folks would leverage a more universal https instead?

In my use case, I was assuming Spec files would be local to the repo (scripted), and updated when we update our dependencies.
https://github.com/dmiluski/CarthageSpecUpdater/blob/master/update_carthage_specs

Why:
I'm tired of maintaining my own Spec repo for binary only dependencies, which don't already offer a Binary Spec json file. Thus trying to automate. Potential for alternate approaches? Also tired of manually downloading, and adding to our repo (bloating).

So, I had lead with the approach of the `overriddenDescription` as being the unique identifier to a transitive absoluteURL that was used for actual file loading?

Reply:
After thinking about it a bit more, I don't think this is an issue.